lib/private.js

```diff
@@ -0,0 +1,39 @@
+'use strict';
+
+var
+    querystring = require('querystring'),
+    {
+        KlaviyoError
+    } = require('./errors.js');
+
+const
+    WRITE_ACCESS_ERROR = 'This value is not writeable.';
+
+class Private {
+    constructor(privateToken)
+    {
+        this._token = privateToken;
+    }
+    
+    get token() {
+        return this._token;
+    }
+
+    set token(value) {
+        throw new KlaviyoError(WRITE_ACCESS_ERROR);
+    }
+
+    prepareV1Data(data={}) {
```

Comment:
I hate these functions, but I also hate the idea of making everything inherit from the request class